src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/AutoConfirmUser/AutomaticallyConfirmOrganizationUserCommand.cs

```diff
@@ -0,0 +1,200 @@
+
+public class AutomaticallyConfirmOrganizationUserCommand(IOrganizationUserRepository organizationUserRepository,
+    IOrganizationRepository organizationRepository,
+    IAutomaticallyConfirmOrganizationUsersValidator validator,
+    IEventService eventService,
+    IMailService mailService,
+    IUserRepository userRepository,
+    IPushRegistrationService pushRegistrationService,
+    IDeviceRepository deviceRepository,
+    IPushNotificationService pushNotificationService,
+    IFeatureService featureService,
+    IPolicyRequirementQuery policyRequirementQuery,
+    ICollectionRepository collectionRepository,
+    ILogger<AutomaticallyConfirmOrganizationUserCommand> logger) : IAutomaticallyConfirmOrganizationUserCommand
+{
+    public async Task<CommandResult> AutomaticallyConfirmOrganizationUserAsync(AutomaticallyConfirmOrganizationUserRequest request)
+    {
+        var requestData = await RetrieveDataAsync(request);
+
+        if (requestData.IsError) return requestData.AsError;
+
+        var validatedData = await validator.ValidateAsync(requestData.AsSuccess);
+
+        if (validatedData.IsError) return validatedData.AsError;
+
+        var validatedRequest = validatedData.Request;
+
+        var successfulConfirmation = await organizationUserRepository.ConfirmOrganizationUserAsync(validatedRequest.OrganizationUser);
+
+        if (!successfulConfirmation) return new None();
+
+        return await validatedRequest.ToCommandResultAsync()
+            .MapAsync(CreateDefaultCollectionsAsync)
+            .MapAsync(LogOrganizationUserConfirmedEventAsync)
+            .MapAsync(SendConfirmedOrganizationUserEmailAsync)
+            .MapAsync(DeleteDeviceRegistrationAsync)
+            .MapAsync(PushSyncOrganizationKeysAsync)
+            .ToResultAsync();
+    }
+
+    private async Task<CommandResult<AutomaticallyConfirmOrganizationUserRequestData>> CreateDefaultCollectionsAsync(
+        AutomaticallyConfirmOrganizationUserRequestData request)
+    {
+        try
+        {
+            if (!featureService.IsEnabled(FeatureFlagKeys.CreateDefaultLocation)
+                || string.IsNullOrWhiteSpace(request.DefaultUserCollectionName)
+                || !(await policyRequirementQuery.GetAsync<OrganizationDataOwnershipPolicyRequirement>(request.UserId))
+                    .RequiresDefaultCollectionOnConfirm(request.Organization.Id))
+            {
+                return request;
+            }
+
+            await collectionRepository.CreateAsync(
+                new Collection
+                {
+                    OrganizationId = request.Organization.Id,
+                    Name = request.DefaultUserCollectionName,
+                    Type = CollectionType.DefaultUserCollection
+                },
+                groups: null,
+                [new CollectionAccessSelection
+                {
+                    Id = request.OrganizationUser.Id,
+                    Manage = true
+                }]);
```

Comment:
💭 (non-blocking) Thoughts on making this an idempotent repository method as well? e.g. `CreateDefaultCollectionAsync`. Given that we are creating it in multiple places and creating default collections properly is quite important. (not this PR though)